File: sdk/python/examples/greedy.py
"""Greedy bot — always moves toward the nearest cheese."""

import random

from pyrat_sdk import Bot, Context, Direction, GameState


class Greedy(Bot):
    name = "Greedy"
    author = "PyRat SDK"

    def think(self, state: GameState, ctx: Context) -> Direction:
        candidates = state.nearest_cheeses()
        if candidates:
            result = random.choice(candidates)
            if result.path:
                ctx.send_info(
                    player=state.my_player,
                    multipv=1,
                    target=result.target,
                    pv=result.path,
                    message=f"target {result.target}, {len(result.path)} steps",
                )
                return result.path[0]
        return Direction.STAY


if __name__ == "__main__":
    Greedy().run()

File: sdk/rust/examples/greedy.rs
use pyrat_sdk::{Bot, Context, Direction, GameState, Options};
use rand::prelude::IndexedRandom;

struct Greedy;

impl Options for Greedy {}

impl Bot for Greedy {
    fn think(&mut self, state: &GameState, ctx: &Context) -> Direction {
        let candidates = state.nearest_cheeses(None);
        let chosen = candidates.choose(&mut rand::rng());
        match chosen {
            Some(result) if !result.path.is_empty() => {
                let target = (result.target.x, result.target.y);
                let mut pos = state.my_position();
                let path: Vec<(u8, u8)> = result
                    .path
                    .iter()
                    .map(|d| {
                        pos = d.apply_to(pos);
                        (pos.x, pos.y)
                    })
                    .collect();
                ctx.send_info(
                    Some(target),
                    0,
                    0,
                    0.0,
                    &path,
                    &format!("target ({}, {}), {} steps", target.0, target.1, path.len()),
                );
                result.path[0]
            },
            _ => Direction::Stay,
        }
    }
}

fn main() {
    pyrat_sdk::run(Greedy, "Greedy", "PyRat");
}
